spec/lib/git_worktree_spec.rb

```diff
@@ -236,4 +237,58 @@ def open_existing_repo
       FileUtils.rm_rf(dirname) if Dir.exist?(dirname)
     end
   end
+
+  context "branches" do
```

Comment:
The `it`s and `context` should reference the method-under-test with the Ruby notation of whether its a class or instance method...So, this should look something like...

```ruby
describe "#branches" do
  it "all branches" do
    expect(test_repo.branches).to match_array(%w(master branch1 branch2))
  end

  it "local branches only" do
    expect(test_repo.branches(:local)).to match_array(%w(master branch1 branch2))
  end

  it "remote branches only" do
    expect(test_repo.branches(:remote)).to be_empty
  end
end

describe "#file_list" do
end

describe "#branch_info" do
end

# etc
```

Since you have some shared lets...you can either wrap the whole thing in a giant context like `context "with the branch_and_tag git repo"` or move them up to the top-level, and just give them better names like `let(:branch_and_tag_repo)`